### ISSUE TITLE
Clone params and params.query before running hooks

### ISSUE THREAD

**Issue by marshallswain** [feathersjs/feathers#675]
Since some hooks modify params and params.query, we should consider making a shallow clone of those objects before passing them down the hook chain.

**Comment by samhunta** [comment]
This would be a breaking change though. What if a feathers plugin purposely intended to modify params / params.query?

**Comment by daffl** [comment]
That is true. I don't think you really should but it would probably make more sense as a plugin (maybe using ImmutableJS for the hook chain).
